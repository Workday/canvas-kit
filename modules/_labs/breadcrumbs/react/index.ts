import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
