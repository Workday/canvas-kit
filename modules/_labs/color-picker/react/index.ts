export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
