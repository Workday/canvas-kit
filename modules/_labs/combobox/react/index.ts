import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
