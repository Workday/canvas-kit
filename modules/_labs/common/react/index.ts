export * from './lib/Flex';
export * from './lib/Box';
