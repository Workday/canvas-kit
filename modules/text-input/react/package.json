{
  "name": "@workday/canvas-kit-react-text-input",
  "version": "3.0.0-alpha.6",
  "description": "Canvas styled text input",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "http://github.com/Workday/canvas-kit/tree/master/modules/text-input/react"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts"
  ],
  "scripts": {
    "watch": "yarn build:es6 -w",
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build": "npm-run-all --parallel build:cjs build:es6"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "text-input"
  ],
  "peerDependencies": {
    "react": ">= 16.8 < 17"
  },
  "dependencies": {
    "@workday/canvas-kit-react-common": "^3.0.0-alpha.6",
    "@workday/canvas-kit-react-core": "^3.0.0-alpha.6",
    "@workday/canvas-kit-react-icon": "^3.0.0-alpha.6",
    "@workday/canvas-system-icons-web": "^1.0.1",
    "emotion": "^9.2.12",
    "react-emotion": "^9.2.12"
  }
}
