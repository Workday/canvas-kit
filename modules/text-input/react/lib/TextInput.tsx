import * as React from 'react';
import {styled, Themeable} from '@workday/canvas-kit-labs-react-core';
import {GrowthBehavior, ErrorType, errorRing} from '@workday/canvas-kit-react-common';
import {borderRadius, inputColors, spacingNumbers, type} from '@workday/canvas-kit-react-core';

export interface TextInputProps
  extends Themeable,
    GrowthBehavior,
    React.InputHTMLAttributes<HTMLInputElement> {
  error?: ErrorType;
  inputRef?: React.Ref<HTMLInputElement>;
}

const Input = styled('input')<Pick<TextInputProps, 'error' | 'grow'>>(
  {
    ...type.body,
    border: `1px solid ${inputColors.border}`,
    display: 'block',
    backgroundColor: inputColors.background,
    borderRadius: borderRadius.m,
    boxSizing: 'border-box',
    height: 40,
    minWidth: 280,
    transition: '0.2s box-shadow, 0.2s border-color',
    padding: spacingNumbers.xxs, // Compensate for border
    margin: 0, // Fix Safari
    '&::placeholder': {
      color: inputColors.placeholder,
    },
    '&:hover': {
      borderColor: inputColors.hoverBorder,
    },
    '&:focus:not([disabled])': {
      borderColor: inputColors.focusBorder,
      boxShadow: `inset 0 0 0 1px ${inputColors.focusBorder}`,
      outline: 'none',
    },
    '&:disabled': {
      backgroundColor: inputColors.disabled.background,
      borderColor: inputColors.disabled.border,
      color: inputColors.disabled.text,
      '&::placeholder': {
        color: inputColors.disabled.text,
      },
    },
    '::-ms-clear': {
      display: 'none',
    },
  },
  ({error}) => ({
    ...errorRing(error),
  }),
  ({grow}) =>
    grow && {
      width: '100%',
    }
);

export default class TextInput extends React.Component<TextInputProps> {
  static ErrorType = ErrorType;

  static defaultProps = {
    inputRef: React.createRef<HTMLInputElement>(),
  };

  render() {
    // TODO: Standardize on prop spread location (see #150)
    const {grow, inputRef, error, ...inputProps} = this.props;

    return <Input type="text" ref={inputRef} grow={grow} error={error} {...inputProps} />;
  }
}
