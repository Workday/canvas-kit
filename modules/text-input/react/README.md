# Canvas Kit Text Input

Canvas-styled React text input components.

## Installation

```sh
yarn add @workday/canvas-kit-react
```

or

```sh
yarn add @workday/canvas-kit-react-text-input
```

# TextInput

## Usage

#### Simple Example

**Note:** While a base input component is provided in this package, it is **not accessible** when
used as is. It should be used in tandem with [`FormField`](../../form-field/react) to be made fully
accessible (see below).

```tsx
import * as React from 'react';
import TextInput from '@workday/canvas-kit-react-text-input';

<TextInput placeholder="Placeholder" value={this.state.value} onChange={() => {}} />;
```

#### Accessible Example

```tsx
import * as React from 'react';
import TextInput from '@workday/canvas-kit-react-text-input';
import FormField from '@workday/canvas-kit-react-form-field';

// Accessible default state
<FormField label="My Field" inputId="my-input-field">
  <TextInput placeholder="Placeholder" value={this.state.value} onChange={() => {}} />;
</FormField>

// Accessible error state
<FormField
  label="My Field"
  inputId="my-input-field"
  hintText="Helpful text to resolve error"
  hintId="my-input-field-error"
  error={FormField.ErrorType.Error}
>
  <TextInput placeholder="Placeholder" value={this.state.value} onChange={() => {}} />;
</FormField>
```

If use inside a FormField doesn't work for your use case, you can use the `aria-labelledBy`
attribute.

```tsx
import * as React from 'react';
import TextInput from '@workday/canvas-kit-react-text-input';
<label id="123">Label</label>
...
<TextInput placeholder="Placeholder" value={this.state.value} onChange={() => {}} aria-labelledBy="123" />;
```

## Static Properties

#### `ErrorType: ErrorType`

```tsx
<TextInput error={TextInput.ErrorType.Alert} />
```

## Component Props

### Required

#### `label: string`

> The label text displayed for the input.

### Optional

#### `disabled: boolean`

> Whether or not the input is disabled.

Default: `false`

---

#### `error: ErrorType`

> The type of error to display, if any.

| Type  | Description                     |
| ----- | ------------------------------- |
| Error | Red outline with error icon.    |
| Alert | Yellow outline with alert icon. |

Default: `undefined`

---

#### `inputRef: React.Ref<HTMLInputElement>`

> The ref callback for the inner input element.

Default: `undefined`

---

#### `onChange: React.ChangeEventHandler<HTMLInputElement>`

> The callback fired when the value is changed.

Default: `undefined`

---

#### `placeholder: string`

> Placeholder text to be displayed if there is no input value.

Default: `''`

---

#### `readOnly: boolean`

> If true, user will be unable to interact with the field.

Default: `false`

---

#### `type: string`

> HTML5 input type

Default: `text`

---

#### `value: any`

> Input value

Default: `undefined`

---
