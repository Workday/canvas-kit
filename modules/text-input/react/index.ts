import InputIconContainer from './lib/InputIconContainer';
import TextInput from './lib/TextInput';

export default TextInput;
export {InputIconContainer, TextInput};
export * from './lib/InputIconContainer';
export * from './lib/TextInput';
