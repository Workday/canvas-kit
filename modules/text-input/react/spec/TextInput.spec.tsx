import * as React from 'react';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TextInput from '../lib/TextInput';

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<TextInput/>);
  expect(asFragment()).toMatchSnapshot();
})


// describe('TextInput', () => {
//   // test('TextInput should spread extra props', () => {
//   //   const component = mount(<TextInput data-propspread="test" onChange={jest.fn()} />);
//   //   const input = component
//   //     .find('input') // TODO: Standardize on prop spread location (see #150)
//   //     .getDOMNode();
//   //   expect(input.getAttribute('data-propspread')).toBe('test');
//   //   component.unmount();
//   // });

//   test('TextInput should grow', () => {
//     // const component = mount(<TextInput grow={true} onChange={jest.fn()} />);
//     // const input = component
//     //   .find('input');
//     //   expect(getComputedStyle(input.getDOMNode()).getPropertyValue('width')).toBe('100%');
//     // component.unmount();

//     const {container} = render(
//       <TextInput grow={true} onChange={jest.fn()} />
//     );

//     const style = 

//     expect(container.firstChild).toHaveAttribute('width', '100%');

//   });
// });



// describe('Text Input Accessibility', () => {
//   test('Text Input in a FormField should pass axe DOM accessibility guidelines', async () => {
//     const html = ReactDOMServer.renderToString(
//       <FormField
//         label="My Field"
//         inputId="my-input-field"
//         hintText="Helpful text to resolve error"
//         hintId="my-input-field-error"
//         error={FormField.ErrorType.Error}
//       >
//         <TextInput placeholder="Placeholder" value={'Hello'} onChange={jest.fn()} />;
//       </FormField>
//     );
//     expect(await axe(html)).toHaveNoViolations();
//   });

//   test('Text Input with `aria-labelledby` should pass axe DOM accessibility guidelines', async () => {
//     const html = ReactDOMServer.renderToString(
//       <>
//         <label id="123">Label</label>
//         <TextInput
//           placeholder="Placeholder"
//           value={'Hello'}
//           aria-labelledby="123"
//           onChange={jest.fn()}
//         />
//         ;
//       </>
//     );
//     expect(await axe(html)).toHaveNoViolations();
//   });
// });
