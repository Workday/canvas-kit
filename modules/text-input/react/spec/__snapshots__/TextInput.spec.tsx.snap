// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  border: 1px solid #7b858f;
  display: block;
  background-color: #ffffff;
  border-radius: 4px;
  box-sizing: border-box;
  height: 40px;
  min-width: 280px;
  -webkit-transition: 0.2s box-shadow,0.2s border-color;
  transition: 0.2s box-shadow,0.2s border-color;
  padding: 8px;
  margin: 0;
}

.emotion-0::-webkit-input-placeholder {
  color: #5e6a75;
}

.emotion-0::-moz-placeholder {
  color: #5e6a75;
}

.emotion-0:-ms-input-placeholder {
  color: #5e6a75;
}

.emotion-0::placeholder {
  color: #5e6a75;
}

.emotion-0:hover {
  border-color: #333d47;
}

.emotion-0:focus:not([disabled]) {
  border-color: #0875e1;
  box-shadow: inset 0 0 0 1px #0875e1;
  outline: none;
}

.emotion-0:disabled {
  background-color: #f6f7f8;
  border-color: #b9c0c7;
  color: #a1aab3;
}

.emotion-0:disabled::-webkit-input-placeholder {
  color: #a1aab3;
}

.emotion-0:disabled::-moz-placeholder {
  color: #a1aab3;
}

.emotion-0:disabled:-ms-input-placeholder {
  color: #a1aab3;
}

.emotion-0:disabled::placeholder {
  color: #a1aab3;
}

<input
    class="emotion-0 emotion-1"
    type="text"
  />
</DocumentFragment>
`;
